### ISSUE TITLE
[5.3] [Queue] queue:listen default value maxTries=0 let every job fail with a MaxAttemptsExceededException

### ISSUE THREAD

**Issue by larsbo** [laravel/framework#15370]
- Laravel Version: 5.3.7
- PHP Version: 7.0.4
- Database Driver & Version: MySQL 5.5.39

### Description:

When I run `queue:listen` it spawns new child processes like this by default:
`artisan queue:work  --once --queue=default --delay=0 --memory=128 --sleep=3 --tries=3 --env=production`

This causes a `MaxAttemptsExceededException` for every single job and make it fail instantly without a single try. If I set `maxTries` to something larger than 0 the job are correctly processed.

**Comment by GrahamCampbell** [comment]
That looks correct.

**Comment by GrahamCampbell** [comment]
Max trys 0 means never process any jobs. 1 means only try once.

**Comment by larsbo** [comment]
Well, the code suggests something else for maxTries = 0:
https://github.com/laravel/framework/blob/5.3/src/Illuminate/Queue/Worker.php#L269
As same as the docs: "If you do not specify a value for the --tries option, jobs will be attempted indefinitely."

**Comment by sisve** [comment]
The linked line of code, and the docs, infer that maxTries=0 should retry indefinitely. The exception you get indicates that the check for ´$maxTries === 0´ failed... so could this be a typing issue where $maxTries is the string "0"?

**Comment by larsbo** [comment]
I logged the queue worker execution and got the following (stripped) result for the `WorkerOptions` parameters given as console arguments:

```php
[object] (Illuminate\Queue\WorkerOptions: {
"delay":"0",
"memory":"128",
"timeout":60,
"sleep":"3",
"maxTries":"0"
},
```
This confirms that every argument except the `timeout`are interpreted as string instead of int.

Can you please review this issue again, @GrahamCampbell ?

**Comment by taylorotwell** [comment]
This was fixed in 5.3.9.

**Comment by larsbo** [comment]
Ah, great! Thank you for the info :+1:
